functional-tests.sh

```diff
@@ -124,7 +124,19 @@ function show()
     fi
 }
 
-function fail()
+function show_on_success()
+{
+    rv="$1"
+    shift
+
+    if [ "$rv" -eq 0 ]; then
+        echo "$@"
+    fi
+
+    return "$rv"
+}
+
+function show_on_failure()
```

Comment:
Looks like both the functions `show_on_success` and `show_on_failure` are doing the same thing, can't we have just one function?

Reply:
One checks for failure and the other checks for success.

Reply:
ah I see.